Sources/_MatchingEngine/Utility/Peek.swift

```diff
@@ -1,6 +1,6 @@
 // MARK: - _Peekable
 
-protocol _Peekable {
+public protocol _Peekable {
```

Comment:
It's only used within this module.   Does it need to be public?

Reply:
Ah, the `Lexer` was conforming before I deleted that. Will remove